internal/services/pubkey_service.go

```diff
@@ -99,6 +109,50 @@ func DeletePubkey(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	pubkey, err := dao.GetPubkeyDao(r.Context())
+	if err != nil {
+		renderError(w, r, payloads.NewInitializeDAOError(r.Context(), "pubkey resource DAO", err))
+		return
+	}
+
+	resources, err := pubkey.UnscopedListByPubkeyId(r.Context(), id)
```

Comment:
the `Unscoped` prefix means this method does not validate the tenant (which would allow any Customer to delete another Customer's pubkeys, which might be quite dangerous), so we should fetch the pubkey itself first and return 404 if not found.
Also, we can now use the `pubkeyDao`

```suggestion
	resources, err := pubkey.UnscopedListByPubkeyId(r.Context(), pubkey.id)
```

Reply:
What do you mean by this suggestion? This variable is for pubkeyDao and does not have an 'id' field.